exp/api/v1beta2/awsmachinepool_types.go

```diff
@@ -52,6 +52,11 @@ type AWSMachinePoolSpec struct {
 	// AvailabilityZones is an array of availability zones instances can run in
 	AvailabilityZones []string `json:"availabilityZones,omitempty"`
 
+	// AvailabilityZoneSubnetType specifies which type of subnets to use when an availability zone is specified.
+	// +kubebuilder:validation:Enum:=public;private
```

Comment:
So that people can explicitly choose that.

Reply:
I can definitely add that. Currently not setting the value does that, but we could add an explicit toggle for it as well in case the default logic changes in the future.

Reply:
@richardcase I've rebased and added an option for `all` along with updated unit tests.

Reply:
Thanks @DavidSpek ...i'll take a look

Reply:
Thanks!